src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -277,8 +289,16 @@ static ColumnWithTypeAndName readColumnWithDate32Data(std::shared_ptr<arrow::Chu
             {
                 Int32 days_num = static_cast<Int32>(chunk.Value(value_i));
                 if (days_num > DATE_LUT_MAX_EXTEND_DAY_NUM || days_num < -DAYNUM_OFFSET_EPOCH)
-                    throw Exception{ErrorCodes::VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE,
-                            "Input value {} of a column \"{}\" is out of allowed Date32 range, which is [{}, {}]", days_num, column_name, DAYNUM_OFFSET_EPOCH, DATE_LUT_MAX_EXTEND_DAY_NUM};
+                {
+                    if (likely(date_time_overflow_mode == DateTimeOverflowMode::SATURATE))
```

Comment:
`likely`/`unlikely` annotations are usually overkill, the branch predictor of the CPU does a good-enough job. Also, if a user happens to set the mode to != SATURATE, then this will become quite expensive. Therefore, please remove `likely`

Besides that, this should really be a `switch` over the enum cases.

Reply:
> please remove likely

Done

> this should really be a switch over the enum cases.

AFAIK there is no such thing as `constexpr switch`, that's why I kept the explicit if-else.